packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1193,13 +1191,40 @@ function workLoop(isYieldy) {
   if (!isYieldy) {
     // Flush work without yielding
     while (nextUnitOfWork !== null) {
-      nextUnitOfWork = performUnitOfWork(nextUnitOfWork);
+      nextUnitOfWork = performUnitOfWork(nextUnitOfWork, isYieldy);
     }
   } else {
     // Flush asynchronous work until there's a higher priority event
     while (nextUnitOfWork !== null && !shouldYieldToRenderer()) {
-      nextUnitOfWork = performUnitOfWork(nextUnitOfWork);
+      nextUnitOfWork = performUnitOfWork(nextUnitOfWork, isYieldy);
+    }
+  }
+}
+
+function beginWorkInDEV(
+  current: Fiber | null,
+  workInProgress: Fiber,
+  renderExpirationTime: ExpirationTime,
+  isYieldy: boolean,
+): Fiber | null {
+  try {
+    return beginWork(current, workInProgress, renderExpirationTime);
+  } catch (thrownValue) {
+    resetContextDependences();
+    resetHooks();
+    if (nextUnitOfWork === null) {
```

Comment:
I don't think there's any way to hit this condition. `nextUnitOfWork` should always be `workInProgress`.